api/src/serge/utils/generate.py

```diff
@@ -59,10 +61,34 @@ async def generate(
 
         try:
             chunk = chunk.decode("utf-8")
+            if(outputLength> 0 ):##if the ai has started responding, log the chunk and length
+                #logger.warn("a chunk: "+chunk)
+                index = (lastchunk+chunk).find("###")##combine the last two chunks to make sure that if the ### was on a border between chunks we still detect it
+                if (index != -1) :##detect the ai responding to itself
+                    #logger.warn("bad response: "+lastchunk+"|"+chunk);
+                    ##find which cunk it was in
+                    if(index<len(lastchunk)):##it was the last chunk
+                        logger.warn("AI responded to itself in the last chunk, cropping response");
+                        lastchunk= lastchunk[:index]##remove everything after the ai's response
+                        yield lastchunk ## return only cropped chunk
+                    else:
+                        logger.warn("AI responded to itself in the current chunk, cropping response");
+                        index -= len(lastchunk)##adjust the index to be in the current chunk
+                        chunk = chunk[:index]##remove everything after the ai's response
+                        yield lastchunk ## return the ok chunk
+                        yield chunk ## return the cropped chunk
+                    #either way kill llama and finish the response
+                    procLlama.kill()## stop llama from continuing to respond
+                    return_code = 0 ##not sure if this is needed
+                    return ## stop the generator
+            else :
+                #logger.warn("u chunk: "+chunk)
+                outputLength += len(chunk) ## incremet the length of the prompt
         except UnicodeDecodeError:
             return
 
-        yield chunk
+        yield lastchunk
+        lastchunk = chunk
```

Comment:
this basically buffers the cunks just so if one chunk ends with ## and the next one starts with # it still gets detected.